nodebuilder/das/daser.go

```diff
@@ -15,6 +15,7 @@ func NewDASer(
 	store header.Store,
 	batching datastore.Batching,
 	fraudService fraud.Module,
-) *das.DASer {
-	return das.NewDASer(da, hsub, store, batching, fraudService)
+	options ...das.Option,
+) (*das.DASer, error) {
+	return das.NewDASer(da, hsub, store, batching, fraudService, options...)
 }
```

Comment:
If this file is removed, then using `das.DASer` in the [module construction](https://github.com/celestiaorg/celestia-node/pull/1225/files#diff-9eef3a4c09b5618e9b60c229170908fc389f8a28ce84e29e47c82d33b74e32b2L20) breaks DI.

The reason whyis: _Quoting @distractedm1nd_

> its because of the casting

> ```
> store header.Store,
>	batching datastore.Batching,
>	fraudService fraud.Module,
> ```
>```
>  getter header.Getter,
>	dstore datastore.Datastore,
>	bcast fraud.Broadcaster,
>```
> we had a discussion about this before (whether to keep such proxy constructors for casting, or to change the original constructor to use what the DI provides) and we decided that its better to keep this ugliness on the DI so that the constructor can take exactly what is needed

Thus we are keeping this file.

Reply:
Ok, thanks for noting that!